### PR TITLE
adding batch write pk insertion test

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
This PR adds new test cases. 
### What is changed and how it works?
In this PR, we generate different schemas with pk. Later, we generate data based on the schema. 
With data and schema, we can perform insert and select with our code. If the data we insert into tikv agreeing with generated data, then we consider our code is working as expected.

### CODE REVIEW

File: core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala [hunk 1]
```diff
@@ -202,6 +202,7 @@ class BaseDataSourceTest(val table: String,
     val df = queryDatasourceTiDBWithTable(sortCol, tableName = tblName)
     val tidbResult = seqRowToList(df.collect(), df.schema)
 
+    println(s"running test on table $tblName")
```
Comment (marsishandsome) [inline core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: use log?

Comment (zhexuany) [reply core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: We use `println` only in our test codebase. It should be ok. Printing this out can help us know whether the test is run or not. In addition, with this info, it helps us identify test failing on which table.

Comment (marsishandsome) [reply core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala]: image

better to use log4j

**Pull request merged**